packages/esm-patient-list-app/src/api/queries.ts

```diff
@@ -113,3 +117,40 @@ export function useRemovePatientsFromOfflinePatientListMutation() {
     }
   });
 }
+
+export function usePatientListDetails(patientListUuid: string): [OpenmrsCohort, KeyedMutator<{ data: OpenmrsCohort }>] {
+  if (patientListUuid) {
+    const {
+      data: cohortDetails,
+      error,
+      mutate,
+    } = useSWR<{ data: OpenmrsCohort }, Error>(`${cohortUrl}/cohort/${patientListUuid}`, openmrsFetch);
+    if (error) {
+      throw error;
+    }
+    return [cohortDetails?.data, mutate];
+  }
+}
+
+interface CohortResponse<T = any> {
+  results: T[];
+  error;
+}
+
+export function usePatientListMembers(
+  patientListUuid: string,
+  startIndex: number = 0,
+  pageSize: number = 10,
+  v: string = 'full',
+): OpenmrsCohortMember[] {
+  if (patientListUuid) {
+    const { data, error, mutate } = useSWR<{ data: CohortResponse<OpenmrsCohortMember> }, Error>(
+      `${cohortUrl}/cohortmember?cohort=${patientListUuid}&startIndex=${startIndex}&limit=${pageSize}&v=${v}`,
+      openmrsFetch,
+    );
+    if (error) {
+      throw error;
+    }
+    return data?.data?.results;
+  }
```

Comment:
The same here, the hook could be reduced to one line.
This would mean that you'd have to write `data?.data?.results` at the call site - if this is something you'd like to avoid, consider the following:

```ts
export function usePatientListMembers(
  patientListUuid: string,
  startIndex: number = 0,
  pageSize: number = 10,
  v: string = 'full',
) {
  const swrResult = useSWR<{ data: CohortResponse<OpenmrsCohortMember> }, Error>(
    `${cohortUrl}/cohortmember?cohort=${patientListUuid}&startIndex=${startIndex}&limit=${pageSize}&v=${v}`,
    openmrsFetch,
  );
  return { ...swrResult, data: swrResult.data?.data?.results };
}
```